src/lib/decorations.typ

```diff
@@ -358,3 +365,78 @@
   // move to end point so the current position after this is the end position
   move-to(end)
 }
+
+
+#let _cetz-content = content
+#let _cetz-anchor = anchor
+
+#let cylinder(
+  start,
```

Comment:
API wise, I would prefer something like:
```
cylinder(pt, anchor: none, name: none, content: none, ..style) = {...}
```

And have width and height as a style: `size: (1,2)` (example).
This way you can easily create lots of shapes with the same size.
Using the `anchor:` argument, you can change the placement relative to `pt`.